feat: Add new selects menus (#1335)
discordjs/guide: guide/interactions/select-menus.md

You can have a maximum of five `ActionRow`s per message, and one select menu within an `ActionRow`.
:::

> The available select menus are: <DocsLink path="class/StringSelectMenuBuilder"/>, <DocsLink path="class/UserSelectMenuBuilder"/>, <DocsLink path="class/RoleSelectMenuBuilder"/>, <DocsLink path="class/ChannelSelectMenuBuilder"/>, <DocsLink path="class/MentionableSelectMenuBuilder"/> classes.

Reviewer: ```suggestion
The available select menus are: <DocsLink path="class/StringSelectMenuBuilder"/>, <DocsLink path="class/UserSelectMenuBuilder"/>, <DocsLink path="class/RoleSelectMenuBuilder"/>, <DocsLink path="class/ChannelSelectMenuBuilder"/>, <DocsLink path="class/MentionableSelectMenuBuilder"/>.
```
`classes` feels wrong here, for one they are internally objects and builders and also that the sentence should end there or classes should be at the front, in the previous sentence there were only two instances in the list and the start of sentence was different

-->
![selectephem](./images/selectephem.png)

> ## Diffrent types
> Next thing, that u should know, is that there are multiple menus like for selecting users, roles, channels and menitionables (users, roles).

Reviewer: ```suggestion
## Various types

 There are multiple menus like for selecting users, roles, channels and menitionables (users and roles).
```

### The interactionCreate event

> To receive a <DocsLink path="class/SelectMenuInteraction"/>, attach an <DocsLink path="class/Client?scrollTo=e-interactionCreate" /> event listener to your client and use the <DocsLink path="class/BaseInteraction?scrollTo=isAnySelectMenu"/> type guard to receive any select menu. Also you can use more direct typeguard like `.iStringSelectMenu()`, `.isUserSelectMenu()`, `.isRoleSelectMenu()`, `.isChannelSelectMenu()`, `.isMentionableSelectMenu()` to recive specific type of select menu:

Reviewer: ```suggestion
To receive a <DocsLink path="class/SelectMenuInteraction"/>, attach an <DocsLink path="class/Client?scrollTo=e-interactionCreate" /> event listener to your client and use the <DocsLink path="class/BaseInteraction?scrollTo=isAnySelectMenu"/> type guard to receive any select menus. You can use more direct typeguard like `.iStringSelectMenu()`, `.isUserSelectMenu()`, `.isRoleSelectMenu()`, `.isChannelSelectMenu()`, `.isMentionableSelectMenu()` to recive specific type of select menu:
```
might be good to link to the methods

	}
});
```
## Getting options

> To get options than user seleted, you can access <DocsLink path="class/StringSelectMenuInteraction?scrollTo=values" /> which is an array of options' values or access a Collection with objects like <DocsLink path="class/RoleSelectMenuInteraction?scrollTo=roles" />, <DocsLink path="class/ChannelSelectMenuInteraction?scrollTo=channels" /> ... 

Reviewer: ```suggestion
To get options that the user seleted, you can access <DocsLink path="class/StringSelectMenuInteraction?scrollTo=values" /> which is an array of option values or access the Collection with objects through <DocsLink path="class/RoleSelectMenuInteraction?scrollTo=roles" />, <DocsLink path="class/ChannelSelectMenuInteraction?scrollTo=channels" />
```
since values is plural we must not use option as plural

```js{6}
const { Events } = require('discord.js');

client.on(Events.InteractionCreate, async interaction => {
	if (!interaction.isUserSelectMenu()) return;

	// This collection can have more users if you have set max values
	const user = interactions.users.first();
	await interaction.reply(`Your favourite user is: ${user.tag}`);
});
```

## Multi-select menus

A select menu is not bound to only one selection; you can specify a minimum and maximum amount of options that must be selected. You can use <DocsLink path="class/SelectMenuBuilder?scrollTo=setMinValues" /> and <DocsLink path="class/SelectMenuBuilder?scrollTo=setMaxValues" /> to determine these values.

> Additionally some options may be default but possible to unselect.

Reviewer: ```suggestion
Default options can be changed by the user.
```
